### ISSUE TITLE
Installation instructions - applying the overlay

### ISSUE THREAD

**Issue by argit2** [jappeace/cut-the-crap#46]
I'm a total noob at nix, installed it just for this package. After a lot of time I finally figured I had to ran nix-build to build the program and it was successful. The program runs if I use its full path, but it doesn't detect ffmpeg since I couldn't follow the install instructions to set up the nix-env.

I have no idea how to apply the overlay. Maybe something changed about nix or how nix deals with haskell packages.

I've tried multiple things, including going to ./nix/overlay and running nix-build since there's a default.nix (which i assume is there to apply the overlay contained in haskellPackages.nix), but I get the error:
` error: expression does not evaluate to a derivation (or a set or list of those)`
Copying neither, either or both .nix in the overlay folder to ~/.config/nixpkgs either gives an error
`error: attribute 'cut-the-crap' in selection path 'cut-the-crap' not found`
or a problem with recursion
`error: infinite recursion encountered, at undefined position`

Am I doing it wrong or is there an actual error going on?
Can you please provide clearer installation instructions for beginners like me?

**Comment by jappeace** [comment]
Hi, 
I see the confusion, the install instructions were outdated because it has become a lot easier these days.
I added [additional instructions](https://github.com/jappeace/cut-the-crap#from-source) and fixed the top level `default.nix` to make it ffmpeg aware. 

Thanks for reaching out!
I have the program also hosted as a service now: http://videocut.org/
I just made that work yesterday!

---

> I have no idea how to apply the overlay. Maybe something changed about nix or how nix deals with haskell packages.

This shouldn't be necessary anymore.
but it involves changing the config in nixpkgs: https://github.com/jappeace/linux-config/blob/work-machine/configuration.nix#L9
But you shouldn't have to do this I think
